lint-staged.config.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+const micromatch = require( 'micromatch' );
+
+/**
+ * Internal dependencies
+ */
+const { plugins } = require( './plugins.json' );
+
+/**
+ * Join and escape filenames for shell.
+ *
+ * @param {Array} files Files to join.
+ *
+ * @return {string} Joined files.
+ */
+const joinFiles = ( files ) => {
+	return files.map( ( file ) => `'${ file }'` ).join( ' ' );
+};
+
+module.exports = {
+	'**/*.js': ( files ) => `npm run lint-js ${ joinFiles( files ) }`,
```

Comment:
Circling back to our previous chat, I was hoping that lint-staged would supported what node does with passing args to commands. For [example](https://nodejs.org/api/child_process.html#child_process_asynchronous_process_creation):

```js
const { spawn } = require('node:child_process');
const ls = spawn('ls', ['-lh', '/usr']);
```

Each argument is passed in an array. But alas it seems lint-staged doesn't support this. I see it has been opened as a feature request: https://github.com/lint-staged/lint-staged/issues/989.